tests/pipelines/test_pipelines_common.py

```diff
@@ -1902,6 +1902,30 @@ def test_StableDiffusionMixin_component(self):
             )
         )
 
+    # @pytest.mark.xfail(condition=not os.getenv("RUN_DDUF_TEST", False), strict=True)
+    # Should consider guarding the test with proper transformers and huggingface_hub versions.
+    def test_save_load_dduf(self):
```

Comment:
@SunMarc I added this simple test. However, it's failing with:

```bash
FAILED tests/pipelines/flux/test_pipeline_flux.py::FluxPipelineFastTests::test_save_load_dduf - AttributeError: 'ZipInfo' object has no attribute '_end_offset'
```

I am on Python 3.10.12. 

@Wauplin is this expected? 

To run:

```bash
pytest tests/pipelines/flux/test_pipeline_flux.py::FluxPipelineFastTests::test_save_load_dduf
```

Reply:
Not expected no :smile: Seems like I'll have to address the "TODO:" earlier than what I thought :disappointed:

Reply:
@sayakpaul fixed it in https://github.com/huggingface/huggingface_hub/pull/2692/commits/ba1e6a4e05070d7a902b4d66db29fc9955150b2c. I don't use the private arg anymore :)

Reply:
@SunMarc I think we would need to merge the commit Lucain is mentioning in your branch `dduf-parser-more-feature`. Otherwise, I still have the same [error](https://github.com/huggingface/diffusers/pull/10037/files#r1874795428) when running the test.

Reply:
done !

Reply:
@SunMarc @Wauplin currently, it's failing with the following:

```bash
FAILED tests/pipelines/flux/test_pipeline_flux.py::FluxPipelineFastTests::test_save_load_dduf - huggingface_hub.errors.DDUFExportError: Missing required 'model_index.json' entry in DDUF file.
```

Reply:
can you try the latest commit from my branch on huggingface hub ?

Reply:
Same.

Reply:
As mentioned [on slack](https://huggingface.slack.com/archives/C0821RXC9T5/p1733832282906659) (private) and in https://github.com/huggingface/huggingface_hub/pull/2692#issuecomment-2531395376 I really think we should post-pone the `pipe.save_pretrained(tmpdir, dduf_file=dduf_filename)` feature and therefore remove this test for now.

Reply:
Yeah I am going to modify the test.

Reply:
@Wauplin this is the current test:

```py
def test_save_load_dduf(self):
    from huggingface_hub import export_folder_as_dduf

    components = self.get_dummy_components()
    pipe = self.pipeline_class(**components)
    pipe = pipe.to(torch_device)
    pipe.set_progress_bar_config(disable=None)

    inputs = self.get_dummy_inputs(device="cpu")
    inputs.pop("generator")
    inputs["generator"] = torch.manual_seed(0)

    pipeline_out = pipe(**inputs).images

    with tempfile.TemporaryDirectory() as tmpdir:
        dduf_filename = f"{pipe.__class__.__name__.lower()}.dduf"
        pipe.save_pretrained(tmpdir, safe_serialization=True)
        export_folder_as_dduf(dduf_filename, folder_path=tmpdir)
        loaded_pipe = self.pipeline_class.from_pretrained(tmpdir, dduf_file=dduf_filename).to(torch_device)

    inputs["generator"] = torch.manual_seed(0)
    loaded_pipeline_out = loaded_pipe(**inputs).images

    assert np.allclose(pipeline_out, loaded_pipeline_out)
```

As you can see, I am now leveraging the `export_folder_as_dduf()` functionality instead of directly serializing DDUF with `save_pretrained()`. With this, it's currently failing:

```bash
FAILED tests/pipelines/flux/test_pipeline_flux.py::FluxPipelineFastTests::test_save_load_dduf - huggingface_hub.errors.DDUFExportError: Invalid DDUF file structure.
```

Is this expected?

Reply:
Sorry about my oversight. The scheduler's config should be `scheduler_config.json` and not `config.json`. 

```py
DDUF_FOLDER_REQUIRED_ENTRIES = {
    # Each folder must contain at least one of these entries
    "config.json",
    "tokenizer_config.json",
    "image_processor.json",
    "scheduler_config.json"
}
```

Reply:
Concretely, this is `git diff`:

```diff
diff --git a/src/huggingface_hub/serialization/_dduf.py b/src/huggingface_hub/serialization/_dduf.py
index 23d40505..20cdcea3 100644
--- a/src/huggingface_hub/serialization/_dduf.py
+++ b/src/huggingface_hub/serialization/_dduf.py
@@ -27,6 +27,7 @@ DDUF_FOLDER_REQUIRED_ENTRIES = {
     "config.json",
     "tokenizer_config.json",
     "image_processor.json",
+    "scheduler_config.json"
 }
```

Reply:
With this change, the test is successfully passing but I am seeing:

```bash
src/diffusers/pipelines/transformers_loading_utils.py:96:55: F821 Undefined name `tmp_generation_config_file`
```

while running `make style && make quality`, which makes sense to me.

Reply:
Cc: @SunMarc for the error ^

Reply:
Fixed quality issue !